### PR TITLE
Fix basecharge unique

### PR DESCRIPTION
**Pull request by mganahl**
add guarantee that `BaseCharge.unique()` returns ordered `BaseCharge`

### CODE REVIEW

File: tensornetwork/block_sparse/charge.py [hunk 1]
```diff
@@ -306,35 +305,56 @@ def intersect(self, other, assume_unique=False, return_indices=False) -> Any:
...
+    return obj
+
   def unique(self,
-             return_index=False,
-             return_inverse=False,
-             return_counts=False) -> Any:
+             return_index: Optional[bool] = False,
+             return_inverse: Optional[bool] = False,
+             return_counts: Optional[bool] = False,
+             sort: Optional[bool] = True) -> Any:
```
Comment (Thenerdstation) [inline tensornetwork/block_sparse/charge.py]: Optional means these arguments can be None, is that what you actually want here?

Comment (mganahl) [reply tensornetwork/block_sparse/charge.py]: Oh, actually no. They shoud be `bool`

Comment (mganahl) [reply tensornetwork/block_sparse/charge.py]: removed the `Optional`.

Comment (mganahl) [reply tensornetwork/block_sparse/charge.py]: turns out that `None` is fine, put `Optional` back

Comment (Thenerdstation) [reply tensornetwork/block_sparse/charge.py]: There's a difference between fine and correct. Let's keep it as just bool.

**Pull request merged**